prepared_test.go

```diff
@@ -0,0 +1,28 @@
+package useragent_test
+
+import (
+	"context"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// This test is done weird way, as we need access to the embed module information
+// that are presented in binary only. So we need to build separate main package and
+// check exit code. We are not checking against exact output, as wtring will differ based on
+// OS, architecture and Go version.
+func TestFromModule(t *testing.T) {
+	ctx, stop := signal.NotifyContext(context.Background())
+	t.Cleanup(stop)
+	cmd := exec.CommandContext(ctx, "go", "run", "testdata/from_module.go")
+	output, err := cmd.Output()
+	assert.NoError(t, err)
+	out := strings.TrimSpace(string(output))
+	assert.NotEmpty(t, out)
+	t.Logf(out)
+	// should look like:
+	// go-useragent/(devel) (Kiwi.com dev) go1.17.3 darwin/amd64
```

Comment:
The test should test the output.

Reply:
I also added tests for `Testing` function and additionally checked that all generated UserAgents are valid.